syntax = "proto3";

package aom.iot.controller;

import "device.proto";

enum Type {
    CHECK_SERVER = 0;
    GET_SERVER_CONFIG = 1;
    PUBLISH_DEVICE_CONFIG = 2;
}

message ServerInfo {
    string user = 1;
    string name = 2;
    string uid = 3;
    double client_count_timestamp= 4;
    uint32 client_count = 5;
}

message DeviceConfig {
    aom.iot.device.DeviceInfo device_info = 1;
    aom.iot.device.Device device = 2;
}

message ControllerMessage {
    Type type = 1;
    double timestamp = 2;
    ServerInfo server_info = 3;
    DeviceConfig device_config = 4;
}
